src/SmartTransactionsController.ts

```diff
@@ -43,37 +42,43 @@ export const DEFAULT_INTERVAL = SECOND * 5;
 export type SmartTransactionsControllerConfig = BaseConfig & {
   interval: number;
   clientId: string;
-  chainId: string;
+  chainId: Hex;
   supportedChainIds: string[];
 };
 
+type FeeEstimates = {
+  approvalTxFees: IndividualTxFees | undefined;
+  tradeTxFees: IndividualTxFees | undefined;
+};
+
 export type SmartTransactionsControllerState = BaseState & {
   smartTransactionsState: {
-    smartTransactions: Record<string, SmartTransaction[]>;
+    smartTransactions: Record<Hex, SmartTransaction[]>;
     userOptIn: boolean | undefined;
     userOptInV2: boolean | undefined;
     liveness: boolean | undefined;
-    fees: {
-      approvalTxFees: IndividualTxFees | undefined;
-      tradeTxFees: IndividualTxFees | undefined;
-    };
+    fees: FeeEstimates;
+    feesByChainId: Record<Hex, FeeEstimates>;
+    livenessByChainId: Record<Hex, boolean>;
   };
 };
 
-export default class SmartTransactionsController extends BaseController<
+export default class SmartTransactionsController extends PollingControllerV1<
   SmartTransactionsControllerConfig,
   SmartTransactionsControllerState
 > {
   public timeoutHandle?: NodeJS.Timeout;
 
   private getNonceLock: any;
 
-  public ethersProvider: any;
+  public ethQuery: EthQuery;
```

Comment:
probably should be private. nbd deal

Reply:
done here https://github.com/MetaMask/smart-transactions-controller/pull/237/commits/2b08ccd27a429deaad6d7e318a613ab5b750626b